more macos fixes

grant2258 opened the pull request:
@sonninnos ive added a platform variable check seems some ci  are missing the platform= and that will cause us issues not taking the setting we set and mame just guesses . Just have to run this to be sure the right platform codes getting called in the makefile.  It could also explain the cache going nuts. I Think we starting to get down to the build issues will take some time and patience though. Hmm it looks like the the platforms are set in the variables above never mind that ill need to set extra settings for osx64 as well. Is strane both are set to osx separated by a cross_compile variable only.

grant2258 committed:
add some information for debugging builds

grant2258 committed:
fix ios-arm64 was commented out out to build locally and debug makefiles

grant2258 committed:
silence empy variable name on tvos

grant2258 committed:
check what this is set too arm builds are really slow

grant2258 committed:
fix last build

grant2258 commented:
That should fix the builds. Its worth noting some arch are set lowercase and other uppercase. Windows env variables arent case sensitive, pretty much everything else is.

sonninnos commented:
Sigh, now Windows and Linux  builds started failing too..

Linux x86 with the stray `x86`, yet x64 works:
```
g++    -MMD -MP -MP -D__LIBRETRO__ -DNDEBUG -DPTR64=1 -DCRLF=2 -DLSB_FIRST -DFLAC__NO_DLL -DPUGIXML_HEADER_ONLY -DNATIVE_DRC=drcbe_x64 -DLUA_COMPAT_ALL -DLUA_COMPAT_5_1 -DLUA_COMPAT_5_2 -I"../../../../../src/osd" -I"../../../../../src/emu" -I"../../../../../src/lib/util" x86 -O2 -m64 -std=c++17 -fPIC -pipe -O3 -fno-strict-aliasing -Wno-unknown-pragmas -Wall -Wcast-align -Wformat-security -Wundef -Wwrite-strings -Wno-conversion -Wno-sign-compare -Wno-error=deprecated-declarations -Wno-error=unused-result -Wno-array-bounds -Wno-error=attributes -Wno-stringop-truncation -Wno-stringop-overflow -m64 -std=c++17 -Woverloaded-virtual -Wvla -Wimplicit-fallthrough -flifetime-dse=1 -Wno-class-memaccess -Wsuggest-override -x c++-header -D__LIBRETRO__ -DNDEBUG -DPTR64=1 -DCRLF=2 -DLSB_FIRST -DFLAC__NO_DLL -DPUGIXML_HEADER_ONLY -DNATIVE_DRC=drcbe_x64 -DLUA_COMPAT_ALL -DLUA_COMPAT_5_1 -DLUA_COMPAT_5_2 -I"../../../../../src/osd" -I"../../../../../src/emu" -I"../../../../../src/lib/util" -o "../../../../libretro/obj/x64/libretro/emu.h.gch" -c "../../../../../src/emu/emu.h"
g++: error: x86: No such file or directory
```

Windows with the stray `x86_64` and same with x86:
```
x86_64-w64-mingw32.static-g++    -MMD -MP -MP -D__LIBRETRO__ -DNDEBUG -DX64_WINDOWS_ABI -DPTR64=1 -DCRLF=3 -DLSB_FIRST -DFLAC__NO_DLL -DPUGIXML_HEADER_ONLY -DNATIVE_DRC=drcbe_x64 -DLUA_COMPAT_ALL -DLUA_COMPAT_5_1 -DLUA_COMPAT_5_2 -DWIN32 -I"../../../../../src/osd" -I"../../../../../src/lib" -I"../../../../../src/lib/util" -I"../../../../../3rdparty" -I"../../../../generated" -I"../../../../../3rdparty/zlib" x86_64 -O2 -m64 -std=c++17 -fPIC -pipe -O3 -fno-strict-aliasing -Wno-unknown-pragmas -Wall -Wcast-align -Wformat-security -Wundef -Wwrite-strings -Wno-conversion -Wno-sign-compare -Wno-error=deprecated-declarations -Wno-error=unused-result -Wno-array-bounds -Wno-error=attributes -Wno-stringop-truncation -Wno-stringop-overflow -Wno-nonnull -Wno-stringop-overread -m64 -std=c++17 -Woverloaded-virtual -Wvla -Wimplicit-fallthrough -Wno-class-memaccess -Wsuggest-override -include /builds/libretro/mame/src/osd/libretro/retroprefix.h -o "../../../../libretro/obj/x64/libretro/src/lib/formats/all.o" -c "../../../../../src/lib/formats/all.cpp"
x86_64-w64-mingw32.static-g++: warning: x86_64: linker input file unused because linking not done
x86_64-w64-mingw32.static-g++: error: x86_64: linker input file not found: No such file or directory
```

Eh and now I can't build locally either because
```
Makefile.libretro:32: *** platform not set.  Stop.
```

Edit: Ok, sorted that platform thing out.

Edit2: And to pile on with the fun factor, just got a report that android build isn't actually working even if the build does not fail:
```
Failed to open libretro core: "/data/data/com.retroarch.ra32/cores/mamearcade_libretro_android.so"
Error(s): dlopen failed: library "libc++_shared.so" not found
```

grant2258 commented:
yea I added a platform to stop the mame builds if it wasnt set.  All that was added to the android builds t adding the path.   It must be changes in lua somewhere that's causing it. add platform=xxx to the makefile line or remove the check.

I added the pull req to fix the windows build this code check will be removed. The problem is the the cpu detection code isint working properly. This is nothing to worry about on on the other other builds just a simple condition to set a variable was off. Its the mac build on this we are looking at and a fix has been put in just now so you can build windows again but its really not an issue.

sonninnos commented:
Wohoo the pipeline is finally green! Any ideas for that android dlopen thing?

grant2258 commented:
Ive never used android for retroarch. I start peeking into it but will probably be the weekend ill manage to get the free time(might get to it before then). I can test this locally so no need to bash the build bot. If you can give me a quick guide to install RA android and where to upload your own cores. Ill test locally instead of bashing the build bot. 

what binary are you needing for testing I can add a static link to libs for android local compile is faster as well

sonninnos commented:
I haven't tested the core myself at all either, and only tested stuff a bit with android in general. The artifacts should be out for grabs, and sooner or later the final result in the usual place when the packaging pipeline passes. Just don't install any Play Store versions but take RA from the website, and it will go smoothly. Installing cores manually via the menu should be possible too.

grant2258 commented:
Well Ive found another issue with the makefile.libretro im going to fix it to use env variables  the setting of flags is silly way of doing it. LIBRETRO_CPU and  LIBRETRO are not needed. Its good th ebuilds are all working.

grant2258 commented:
I installed RA some folders done work to browse files you have to just guess one that works. I enabled the logging even set the directory but its not even logging. I cant see any mame0251 bin I need to check that bin to see if its a static or shared. It looks to me like the old one was shared as well from looking at the code.
